src/Lykke.Job.BlockchainOperationsExecutor/Workflow/Sagas/TransactionExecutionSaga.cs

```diff
@@ -158,18 +216,39 @@ private async Task Handle(TransactionBroadcastedEvent evt, ICommandSender sender
 
             if (_stateSwitcher.Switch(aggregate, evt))
             {
-                sender.SendCommand
-                (
-                    new ReleaseSourceAddressLockCommand
-                    {
-                        OperationId = aggregate.OperationId,
-                        TransactionId = aggregate.TransactionId,
-                        BlockchainType = aggregate.BlockchainType,
-                        FromAddress = aggregate.FromAddress
-                    },
-                    Self
-                );
-
+                if (_blockchainSettingsProvider.GetIsExclusiveWithdrawalsRequired(aggregate.BlockchainType))
```

Comment:
Better to make a decision according to aggregate state, but not settings. The only place where settings should be read - is the start of the workflow.